Neos.Flow/Tests/Unit/Mvc/Routing/RouterTest.php

```diff
@@ -405,6 +409,13 @@ public function routeLoadsRoutesConfigurationFromConfigurationManagerIfNotSetExp
     {
         /** @var Router|\PHPUnit\Framework\MockObject\MockObject $router */
         $router = $this->getAccessibleMock(Router::class, ['dummy']);
+        $mockReflectionService = $this->getMockBuilder(ReflectionService::class)->disableOriginalConstructor()->getMock();
+        $mockReflectionService->expects(self::once())->method('getAllSubClassNamesForClass')->with(AbstractController::class)->willReturn([RouteAnnotatedController::class]);
```

Comment:
This test fails because it is still missing a couple of methods to be mocked on the ReflectionService, like `getMethodsAnnotatedWith`, `isMethodPublic` and `getMethodAnnotation` - I'd suggest going with Bastians refactoring to move the configuration stuff out of the `Router`. Then test the configuration building from Reflection in isolation with a Functional test.